src/feature/gui/gui-runner.c

```diff
@@ -656,6 +656,10 @@ void mGUIRun(struct mGUIRunner* runner, const char* path) {
 				runner->core->reset(runner->core);
 				break;
 			case RUNNER_SAVE_STATE:
+				struct mGUIBackground* gbaBackground = (struct mGUIBackground*) stateSaveMenu.background;
+				// If we are saving state, then the screenshot stored for the state previously should no longer be considered up-to-date.
+				// Therefore, mark it as stale, so that at draw time, we will re-draw to the new save state's screenshot.
+				gbaBackground->screenshotId |= SCREENSHOT_INVALID;
```

Comment:
You didn't remove this variable declaration, which can just be merged into the statement anyway.

```suggestion
				// If we are saving state, then the screenshot stored for the state previously should no longer be considered up-to-date.
				// Therefore, mark it as stale, so that at draw time, we will re-draw to the new save state's screenshot.
				((struct mGUIBackground*) stateSaveMenu.background)->screenshotId |= SCREENSHOT_INVALID;
```